src/khoj/routers/api_chat.py

```diff
@@ -590,9 +590,7 @@ async def send_rate_limit_message(message: str):
         )
 
         if compiled_references:
-            headings = "\n- " + "\n- ".join(
-                set([" ".join(c.get("compiled", c).split("Path: ")[1:]).split("\n ")[0] for c in compiled_references])
-            )
+            headings = "\n- " + "\n- ".join(set([c.get("compiled", c).split("\n")[0] for c in compiled_references]))
```

Comment:
Why don't we add the `filename` as a property in the reference and use that instead? In `api.py`, when `compiled_references` is being formed just before the return statement.

Reply:
We're using the first line of the heading because it also contains the heading ancestry of the note, at least when org-mode files are referenced. This granularity would be useful in long org-mode files with lots of notes.